src/app/material-impl/components/test-material/test-material.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { DialogHwComponent } from '../dialog-hw/dialog-hw.component';
+ 
+export interface DialogData {
+  name: string
+}
+@Component({
+  selector: 'app-test-material',
+  templateUrl: './test-material.component.html',
+  styleUrl: './test-material.component.scss'
+
+})
+export class TestMaterialComponent {
+  name: string = '';
+  constructor(public dialog: MatDialog) {}
+  openDialog() {
+    const dialogReference = this.dialog.open(DialogHwComponent, {
+      data: {name: this.name},
```

Comment:
Cuida los espacios en blanco en tu código, punto y coma al final de cada sentencia, identación, eso habla de su limpieza y es importante cuando se trabaja en equipo, te recomiendo instales en tu vscode o en tu proyecto prettier para dar formato automático o por medio de un comando a tu proyecto y no se te pasen.

Reply:
La forma correcta de declarar un objeto es: 
{
  data: { name: this.name }
}